pkg/naming/metrics_query.go

```diff
@@ -88,7 +90,7 @@ type queryPart struct {
 func (q *metricsQuery) Build(series string, resource schema.GroupResource, namespace string, extraGroupBy []string, metricSelector labels.Selector, names ...string) (prom.Selector, error) {
 	queryParts := q.createQueryPartsFromSelector(metricSelector)
 
-	if namespace != "" {
+	if q.namespaced && namespace != "" {
```

Comment:
we don't want to add the check here as it would enable this feature for the other metrics API